.github/workflows/build.yml

```diff
@@ -37,3 +39,50 @@ jobs:
       - name: Run tests
         working-directory: ./v2
         run: go test ./...
+
+  create_release:
+    if: startsWith(github.ref, 'refs/tags/v')
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    needs: [lint, test]
+    steps:
+      - uses: actions/checkout@v4
+      - name: build and release
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          gh release create ${{ github.ref_name }}
+  build_release:
+    if: startsWith(github.ref, 'refs/tags/v')
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    needs: [lint, test, create_release]
+    strategy:
+      matrix:
+        goos: [linux, darwin]
+        goarch: ["386", amd64, arm64]
```

Comment:
I'm not familiar with the `386` ARCH but do we need it?

Reply:
it's the 86 part of x86/64 -- IE 32 bit systems. Still usable until 32 bit integers roll over in 2038.